src/components/expanded-state/ContactProfileState.js

```diff
@@ -1,76 +1,48 @@
 import lang from 'i18n-js';
-import React, { useCallback, useMemo, useState } from 'react';
+import React, { useCallback, useState } from 'react';
 import { Keyboard } from 'react-native';
 import { useNavigation } from '../../navigation/Navigation';
-import { useTheme } from '../../theme/ThemeContext';
 import { magicMemo } from '../../utils';
 import ProfileModal from './profile/ProfileModal';
 import useExperimentalFlag, {
   PROFILES,
 } from '@rainbow-me/config/experimentalHooks';
 import { maybeSignUri } from '@rainbow-me/handlers/imgix';
-import {
-  removeFirstEmojiFromString,
-  returnStringFirstEmoji,
-} from '@rainbow-me/helpers/emojiHandler';
 import {
   useAccountSettings,
   useContacts,
   useENSAvatar,
   usePersistentDominantColorFromImage,
+  useRainbowProfile,
 } from '@rainbow-me/hooks';
-import {
-  addressHashedColorIndex,
-  addressHashedEmoji,
-} from '@rainbow-me/utils/profileUtils';
-
-const ContactProfileState = ({ address, color, contact, ens, nickname }) => {
+const ContactProfileState = ({ address, ens, contactNickname }) => {
   const profilesEnabled = useExperimentalFlag(PROFILES);
-  const contactNickname = contact?.nickname || nickname;
+  const [nickname, setNickname] = useState(contactNickname || ens || '');
   const { goBack } = useNavigation();
   const { onAddOrUpdateContacts } = useContacts();
-  const { colors } = useTheme();
-
-  const [value, setValue] = useState(
-    profilesEnabled
-      ? contactNickname
-      : removeFirstEmojiFromString(contactNickname)
-  );
-
-  const emoji = useMemo(
-    () =>
-      profilesEnabled
-        ? addressHashedEmoji(address)
-        : returnStringFirstEmoji(contactNickname),
-    [address, contactNickname, profilesEnabled]
-  );
-
-  const colorIndex = useMemo(
-    () => (profilesEnabled ? addressHashedColorIndex(address) : color || 0),
-    [address, color, profilesEnabled]
-  );
 
   const { network } = useAccountSettings();
 
+  const { rainbowProfile } = useRainbowProfile(address);
```

Comment:
here's another instance of `useRainbowProfile` where `address` needs to be defined but probably since this is a js file lint is not screaming?